Survival-DEV/blog-backend
# Blog-24: Fix adding comment children

### :dart:  What is been done in this PR:
**endpoint** : `api/comments`
- [x] Add Children to commentEnitty via `create` service instead of pgadmin
- [x] remove  `eager`/ auto-reload /` INNER JOIN` for blogEntity of each comment, and replace it with `blog_id` instead
- [x] clean up the comments enity
- [x] validate the dude ~ partially 

**Relates to #24**

----
### Resquest:
 > 
 ```json
  {
    "content": "YUP YUP ",
    "blog_id": "e7a494be-997e-4e1a-aaa3-f5fdc6b72627",
    "parent_id": "8eb70649-8a3e-4a7d-92cd-b2ad6ec01589"
  }

```
### Response:
 > 
 ```json
[
     {
        "id": "8eb70649-8a3e-4a7d-92cd-b2ad6ec01589",
        "created_at": "2021-08-07T10:26:57.575Z",
        "updated_at": "2021-08-07T10:26:57.575Z",
        "content": "Such alovely DAYYY!",
        "level": 2,
        "children": [
            {
                "id": "9ae3a296-3741-452a-bdb0-99fcb23b0980",
                "created_at": "2021-08-07T10:28:00.276Z",
                "updated_at": "2021-08-07T10:28:00.276Z",
                "content": "YUP YUP ",
                "level": 2,
                "children": []
            }
        ]
    }
]
 
```
### :compass:   How to test it
Steps to test my PR:
1. `npm run ci`
2. make sure to setup your `DATABASE_URL`, `NODE_ENV`, `logDB` & `PORT` in your `.env` file
6. `npm run start:dev`
4. the active routes are-with their CRUD ops- :
   - `api/comments`
